server/routes/prisonerChangeIncentiveLevelDetails.ts

```diff
@@ -0,0 +1,142 @@
+      const systemToken = await hmppsAuthClient.getSystemClientToken(res.locals.user.username)
+      const prisonApi = new PrisonApi(systemToken);
+      const incentivesApi = new IncentivesApi(systemToken)
+
+      try {
+        const prisonerDetails = await prisonApi.getFullDetails(prisonerNumber, true)
+        const {agencyId, bookingId, firstName, lastName, assignedLivingUnit} = prisonerDetails
+        const locationId: string | undefined = assignedLivingUnit?.description
+        const incentiveSummary = await incentivesApi.getIncentiveSummaryForPrisoner(prisonerNumber)
+        const nextReviewDate = incentiveSummary?.nextReviewDate && moment(incentiveSummary.nextReviewDate, 'YYYY-MM-DD HH:mm')
+
+        return res.render('pages/prisonerChangeIncentiveLevelConfirmation.njk', {
+          agencyId,
+          bookingId,
+          breadcrumbPrisonerName: putLastNameFirst(firstName, lastName),
+          cancelUrl: '/',
+          incentiveSummary,
+          manageIncentivesUrl:
+              agencyId && locationId && locationId.includes('-')
+                  ? `/incentive-summary/${agencyId}-${locationId.split('-')[0]}`
+                  : '/',
+          nextReviewDate: nextReviewDate?.format('D MMMM YYYY'),
+          prisonerNumber,
+          prisonerName: formatName(firstName, lastName),
+          profileUrl: `${res.app.locals.dpsUrl}/prisoner/${prisonerNumber}`,
+        })
+      } catch (error) {
+        res.locals.redirectUrl = `incentive-level-details/prisoner/${prisonerNumber}`
+        throw error
+      }
+    }
+
+
+  // @ts-ignore
+  const index = async (req, res) => renderTemplate(req, res, undefined);
+  // @ts-ignore
+  const postForm = async (req, res) => {
+    const offenderNo  = req.params.offenderNo;
+    const errors = [];
+
+    const {agencyId, bookingId, iepLevel, newIepLevel, reason} = req.body || {};
+
+
+    if (!newIepLevel) {
+      errors.push({text: 'Select an incentive level, even if it is the same as before', href: '#newIepLevel'})
+    }
+
+    if (!reason) {
+      errors.push({text: 'Enter a reason for recording', href: '#reason'})
+    }
+
+    if (reason && reason.length > 240) {
+      errors.push({text: 'Comments must be 240 characters or less', href: '#reason'})
+    }
+
+    // TODO: CHECK URL
+    if (errors.length > 0) {
+      return renderTemplate(req, res, {errors, formValues: {newIepLevel, reason}});
+    }
+
+    try {
+      // const changedIncentiveLevel = await incentivesApi.updateIncentiveLevelForPrisoner(prisonerNumber, {
+      //   iepLevel: newIepLevel,
+      //   comment: reason,
+      // })
+      // raiseAnalyticsEvent(
+      //   'Update Incentive level',
+      //   `Level changed from ${iepLevel} to ${newIepLevel} at ${agencyId}`,
+      //   'Incentive level change'
+      // )
+      return renderConfirmation(req, res)
+    } catch (error) {
+      // TODO: CHECK URL
+      res.locals.redirectUrl = `/prisoner/${offenderNo}`
```

Comment:
this will need to redirect to the new prisoner profile on DPS so we would need a whole new environment variable like `PRISONER_PROFILE_URL: "https://prisoner-dev.digital.prison.service.justice.gov.uk"`

Reply:
i just noticed you have https://github.com/ministryofjustice/hmpps-incentives-ui/pull/678/files#diff-7c6cd32da5f152ddbb8d8312c23cd3b5f1e15ee37e3d706ca81344ff6003eebdR210 which is similar and uses the old url (which works cos it's redirected) but maybe both ought to use the new urls…?